src/test/unit/lib/model/InboundThirdpartyTransactionModel.test.js

```diff
@@ -0,0 +1,75 @@
+/**************************************************************************
+ *  (C) Copyright ModusBox Inc. 2019 - All rights reserved.               *
+ *                                                                        *
+ *  This file is made available under the terms of the license agreement  *
+ *  specified in the corresponding source code repository.                *
+ *                                                                        *
+ *  ORIGINAL AUTHOR:                                                      *
+ *       Sridhar Voruganti - sridhar.voruganti@modusbox.com               *
+ **************************************************************************/
+'use strict';
+
+// we use a mock standard components lib to intercept and mock certain funcs
+jest.mock('@mojaloop/sdk-standard-components');
+jest.mock('redis');
+
+const defaultConfig = require('./data/defaultConfig');
+const { Logger, Transports } = require('@internal/log');
+const ThirdpartyTrxnModelIn = require('@internal/model').InboundThirdpartyTransactionModel;
+const mockAuthorizationArguments = require('./data/mockAuthorizationArguments');
+const { MojaloopRequests } = require('@mojaloop/sdk-standard-components');
+const { BackendRequests } = require('@internal/requests');
+const Cache = require('@internal/cache');
+
+describe('inboundThirdpartyTransactionModel', () => {
+    let config;
+    let mockAuthReqArgs;
+    let logger;
+
+    beforeAll(async () => {
+        const logTransports = await Promise.all([Transports.consoleDir()]);
+        logger = new Logger({
+            context: { app: 'inboundThirdpartyTransactionModel-unit-tests' },
+            space: 4,
+            transports: logTransports,
+        });
+    });
+
+    beforeEach(async () => {
+        config = JSON.parse(JSON.stringify(defaultConfig));
+        mockAuthReqArgs = JSON.parse(JSON.stringify(mockAuthorizationArguments));
+    });
+
+    describe('authorizations', () => {
+        let model;
+        let cache;
+
+        beforeEach(async () => {
+            BackendRequests.__getSignedChallenge = jest.fn().mockReturnValue(Promise.resolve(mockAuthReqArgs.internalSignedChallengeResponse));
+            cache = new Cache({
+                host: 'dummycachehost',
+                port: 1234,
+                logger,
+            });
+            await cache.connect();
+
+            model = new ThirdpartyTrxnModelIn({
+                ...config,
+                cache,
```

Comment:
there is no need to used full cache with mocked redis-> instead mock only used, by your code, cache's methods

Reply:
Fixed it